zingolib/src/lightclient.rs

```diff
@@ -504,29 +491,33 @@ pub mod propose;
 
 // other functions
 impl LightClient {
-    /// TODO: Add Doc Comment Here!
+    /// a helper function to unlock the wallet state from whatever type of Arc it is in
+    /// dont worry that &self isnt mut. it contains a Mutex
     #[cfg(not(feature = "sync"))]
-    pub async fn clear_state(&self) {
-        // First, clear the state from the wallet
-        self.wallet.clear_all().await;
-
-        // Then set the initial block
-        let birthday = self.wallet.get_birthday().await;
-        self.set_wallet_initial_state(birthday).await;
-        debug!("Cleared wallet state, with birthday at {}", birthday);
+    pub async fn wallet_mut(&self) -> &LightWallet {
```

Comment:
There's no unlocking going on in the not(sync) version, this is granting immutable access only.

Reply:
hmm, yeah, i noticed this, but when i didnt get compiler errors i assumed that the Locks worked themselves out...